### ISSUE TITLE
Add Non-linear activation functions to PyTorch Frontend

### ISSUE THREAD

**Issue by djl11** [unifyai/ivy#21]
Add [Non-linear activation functions](https://pytorch.org/docs/stable/nn.functional.html#non-linear-activation-functions) to PyTorch frontend:

- [x] #4039
- [x] #4040
- [ ] #2457 
- [ ] #2667 
- [ ] hardtanh
- [ ] hardtanh_
- [ ] hardswish
- [x] #4083
- [x] #4084
- [x] #4085
- [x] #4105
- [x] #4106
- [x] #2723 
- [ ] #2724 
- [x] #4219
- [ ] #4220
- [ ] #4221
- [ ] glu
- [ ] gelu
- [x] #3889
- [ ] hardshrink
- [ ] tanhshrink
- [ ] softsign
- [ ] #3007
- [x] #3966
- [x] #2725 
- [ ] softshrink
- [ ] gumbel_softmax
- [ ] log_softmax
- [x] #2686 
- [x] #2480 
- [ ] #4263
- [ ] silu
- [ ] #3499
- [ ] batch_norm
- [ ] group_norm
- [ ] instance_norm
- [ ] #2947
- [ ] local_response_norm
- [ ] normalize

**Comment by AymenSe** [comment]
relu

**Comment by pritam1322** [comment]
sigmoid #2480

**Comment by AymenSe** [comment]
relu_ leaky_relu leaky_relu_ #2667

**Comment by bsovdat** [comment]
#2686

**Comment by AymenSe** [comment]
leaky_relu_ #2724
leaky_relu #2723
relu_ #2667

**Comment by mjschock** [comment]
- [ ] #2947

**Comment by ashmitkx** [comment]
[] #3007

**Comment by josehamess** [comment]
#3259

**Comment by rana-shahroz** [comment]
- [ ] #3303

**Comment by rana-shahroz** [comment]
- [ ] #3303

**Comment by rana-shahroz** [comment]
- [ ] #3303

**Comment by aryankhatana01** [comment]
- [ ] #3499

**Comment by mibo8** [comment]
- [ ] #3889

**Comment by irmakaras** [comment]
- [ ] #3966

**Comment by hirwa-nshuti** [comment]
- [ ] #4039

**Comment by hirwa-nshuti** [comment]
- [ ] #4040

**Comment by hirwa-nshuti** [comment]
- [ ] #4083

**Comment by hirwa-nshuti** [comment]
- [ ] #4084

**Comment by hirwa-nshuti** [comment]
- [ ] #4085

**Comment by hirwa-nshuti** [comment]
- [ ] #4105

**Comment by hirwa-nshuti** [comment]
- [ ] #4106

**Comment by hirwa-nshuti** [comment]
- [ ] #4219

**Comment by hirwa-nshuti** [comment]
- [ ] #4220

**Comment by hirwa-nshuti** [comment]
- [ ] #4221

**Comment by y-h-chan** [comment]
- [ ] #4263

**Comment by Aboelenien** [comment]
batch_norm #4371

**Comment by hirwa-nshuti** [comment]
- [ ] #4397

**Comment by hirwa-nshuti** [comment]
- [ ] #4398

**Comment by hirwa-nshuti** [comment]
- [ ] #4401

**Comment by hirwa-nshuti** [comment]
- [ ] #4402

**Comment by hirwa-nshuti** [comment]
- [ ] #4403

**Comment by hirwa-nshuti** [comment]
- [ ] #4491

**Comment by hirwa-nshuti** [comment]
- [ ] #4492

**Comment by hirwa-nshuti** [comment]
- [ ] #4494

**Comment by hirwa-nshuti** [comment]
- [ ] #4650

**Comment by hirwa-nshuti** [comment]
- [ ] #4651

**Comment by hirwa-nshuti** [comment]
- [ ] #4652

**Comment by hirwa-nshuti** [comment]
- [ ] #4823

**Comment by TAUIL-Abd-Elilah** [comment]
- [ ] #5797

**Comment by VikramxD** [comment]
- [ ] #6223

**Comment by SR42** [comment]
- [] #6537

**Comment by aryankhatana01** [comment]
- #6552

**Comment by satani99** [comment]
- [ ] #9063

**Comment by bahaozsahin** [comment]
#10201

**Comment by bahaozsahin** [comment]
gumbel_softmax #10201

**Comment by AnnaTz** [comment]
- [ ] #10201

**Comment by Immortalsoulslayer** [comment]
gumbel_softmax #10718

**Comment by Szeryf19** [comment]
- [ ] #14420

**Comment by roudik** [comment]
- [ ] #14660

**Comment by Bright-in** [comment]
#17136

**Comment by Bright-in** [comment]
#17136

**Comment by leonjovanovic** [comment]
Is https://github.com/unifyai/ivy/issues/14660 open to work on or is it taken?

**Comment by AnnaTz** [comment]
> Is #14660 open to work on or is it taken?

I think you can link your PR to the issue and work on it. Whoever opened it ended up adding another function instead of `tanh`.

**Comment by leonjovanovic** [comment]
I think tanh is already done. [Link](https://github.com/unifyai/ivy/blob/770111bdff2c7c010b01965f7da8f63110fd86b6/ivy/functional/frontends/torch/nn/functional/non_linear_activation_functions.py#L115)

I ll do gumbel_softmax [PR](https://github.com/unifyai/ivy/pull/17586)

**Comment by ipsitmantri** [comment]
I want to do layernorm. Can you assign it to me?

**Comment by sahartek** [comment]
- [layer_norm] #17794

**Comment by in-odds** [comment]
#17842

**Comment by faiqali1** [comment]
All the functions are complete. 
Mark this as closed?

**Comment by AnnaTz** [comment]
> All the functions are complete. Mark this as closed?

There is one function remaining.

**Comment by AnnaTz** [comment]
> #17586 `gumbel_softmax` has been merged? any other issues left?

Oh it wasn't linked to the list. Then I'll close this. Thanks!
